### PR TITLE
[ioctl] build did gethash command line into new ioctl

### LINKED ISSUE #3291: [ioctl] build did gethash command line into new ioctl
### What would you like to be added
 - Step 1: create `didgethash.go` and `didgethash_test.go` files in `./ioctl/newcmd/did`. create this directory if not exist. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/did/didgethash.go in `didgethash.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/did/didgethash.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `didgethash` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3291 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewDidGetHashCmd

**Test Configuration**:
- Firmware version: Windows 10 (WSL Version: Ubuntu 18.04)
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
@huof6829 
I'm encounter a problem with writting unittest, my code keep failing with the error message "did does not exist", is there any way to solve this without duplicate the test code inside `didregister` command?

**Comment by huof6829** [comment]
> @huof6829 I'm encounter a problem with writting unittest, my code keep failing with the error message "did does not exist", is there any way to solve this without duplicate the test code inside `didregister` command?

which test function?  I will try.

**Comment by LuckyPigeon** [comment]
@huof6829 
The test function is inside `didgethash_test.go`, named `TestNewDidGetHashCmd`.
My first thought is using `NewDidRegisterCmd` to register a new `did`, but it seems doesn't work well.

**Comment by huof6829** [comment]
> @huof6829 The test function is inside `didgethash_test.go`, named `TestNewDidGetHashCmd`. My first thought is using `NewDidRegisterCmd` to register a new `did`, but it seems doesn't work well.

OK. I will try out.

**Comment by huof6829** [comment]
@LuckyPigeon   the length  of data should be inter-multiple of 32 bytes. Otherwise `abi.Unpack` runs error

File: ioctl/newcmd/did/didgethash.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			abi, err := abi.JSON(strings.NewReader(DIDABI))
+			if err != nil {
+				return errors.Wrap(err, "failed to read abi")
+			}
+			bytecode, err := encodeGet(abi, _getHashName, args[1])
+			if err != nil {
+				return errors.Wrap(err, "invalid bytecode")
+			}
+
+			result, err := action.Read(client, addr, "0", bytecode, contract, 100000)
```
Comment (huof6829) [inline ioctl/newcmd/did/didgethash.go:65]: `100000` is gas limit default value?

File: ioctl/newcmd/did/didgethash.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		},
+	}
+	return cmd
+}
+
+func encodeGet(abi abi.ABI, method, did string) (ret []byte, err error) {
+	_, exist := abi.Methods[method]
+	if !exist {
+		return nil, errors.New("method is not found")
+	}
```
Comment (huof6829) [inline ioctl/newcmd/did/didgethash.go:92]: done by inner of `abi.Pack`.

File: ioctl/newcmd/did/didgethash.go (hunk 1 above)
Comment (LuckyPigeon) [reply ioctl/newcmd/did/didgethash.go:65]: Yes, check [this](https://github.com/iotexproject/iotex-desktop-wallet/issues/381) for more info.

**Review by huangzhiran — APPROVED** [review]

File: ioctl/newcmd/did/didgethash.go (hunk 1 above)
Comment (huof6829) [reply ioctl/newcmd/did/didgethash.go:65]: The value is 
https://github.com/iotexproject/iotex-core/blob/49ce6130a40a48838ca4ffbc4757763b6931633a/ioctl/cmd/action/action.go#L52

**Review by huof6829 — APPROVED** [review]

**Pull request merged**